Signing image builds

zvonkok opened the pull request:
- Enhanced the e2e test
- Added Connaisseur for e2e test
- New Changes: Simple workflow to sign images and push to a registry, 
- Overriding states via ConfigMap
- Added support for  helm pre-install and  helm post-install hooks
- Templates in the CR

Commit:
Enhanced the e2e test undeploy func

Commit:
Merge branch 'master' into sign-image

Commit:
Enhanced the e2e test undeploy func

Commit:
Merge branch 'sign-image' of https://github.com/openshift-psap/special-resource-operator into sign-image

Commit:
Merge branch 'master' into sign-image

Commit:
Added image signature service

Commit:
Merge branch 'master' into sign-image

zvonkok commented on test/undeploy/undeploy.go:
```diff
@@ -36,6 +37,10 @@ func main() {
 
 	cl := framework.NewControllerRuntimeClient()
 
+	framework.DeleteAllSpecialResources(cl)
```
@dagrayvid

dagrayvid commented on test/framework/create_from_yaml.go:
```diff
@@ -41,7 +42,11 @@ func NewControllerRuntimeClient() client.Client {
 	})
```
```suggestion
	       MetricsBindAddress: "0",
	})
```

I think that this change will disable metrics for this controller, which will help us avoid the test flake of
```2021-06-29T13:33:29.011-0400	INFO	exit  	OnError: framework.NewControllerRuntimeClient[create_from_yaml.go:42] unable to start manager: error listening on :8080: listen tcp :8080: bind: address already in use```

zvonkok replied:
This change is already in master. Resolving.

Commit:
Adding MultiBuild

Commit:
Adding MultiBuild

Commit:
Merge Master

Commit:
Added debug.md and multi-build project

Commit:
Merge branch 'master' into sign-image

dagrayvid commented on docs/debug.md:
````diff
@@ -0,0 +1,78 @@
+following:
+
+```bash
+VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make assets
````
```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make assets
```

dagrayvid commented on docs/debug.md:
````diff
@@ -0,0 +1,78 @@
+To update the CR one can run:
+
+```bash
+VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make
````
```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make
```

dagrayvid commented on docs/debug.md:
```diff
@@ -0,0 +1,78 @@
+
+For this to work one has to create a `kustomization.yaml` that will be processed
+during a make run. One cannot just override one state all states are either
+read from one resource. Here is an `kustomization.yaml` as an example from the
```
```suggestion
during a make run. One cannot just override one state, all states are
read from one source (http, local, or ConfigMap). Here is an `kustomization.yaml` as an example from the
```
Found this sentence confusing at first.

Commit:
Update docs/debug.md

Co-authored-by: David Gray <40244437+dagrayvid@users.noreply.github.com>

Commit:
Update docs/debug.md

Co-authored-by: David Gray <40244437+dagrayvid@users.noreply.github.com>

Commit:
Update docs/debug.md

Co-authored-by: David Gray <40244437+dagrayvid@users.noreply.github.com>

SchSeba commented on docs/debug.md:
```diff
@@ -0,0 +1,78 @@
+metadata:
+  name: multi-build
+spec:
+  debug: false
```
I think this needs to be true in this example

SchSeba commented on docs/debug.md:
````diff
@@ -0,0 +1,78 @@
+VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make assets
+```
+
+This command will create a CM with all the states and SRO will use them in the
````
is this only for development or can be used in production also?
as the partner may not want to push the is assets into the SRO repo

Commit:
Merged k8s support

Commit:
Make linter happy
Updating k8s workflow
Added missing file
Added NFD to workflow
Update on RB
Updated manifests generation
Update k8s-e2e.yaml
Testing Concurrency

Commit:
Merge branch 'master' into sign-image

Commit:
Testing Concurrency

Commit:
Merge branch 'sign-image' of https://github.com/openshift-psap/special-resource-operator into sign-image

Commit:
Update Action

Commit:
Merge branch 'master' into sign-image

Commit:
Update manifests for k8s

Commit:
Merge branch 'sign-image' of https://github.com/openshift-psap/special-resource-operator into sign-image

Commit:
Setting SHEL explicitly since on Ubuntu /bin/sh is linked to dash rather then bash

dagrayvid commented on manifests-k8s/0000_namespace_openshift-special-resource-operator.yaml:
```diff
@@ -0,0 +1,7 @@
+  labels:
+    control-plane: controller-manager
+    openshift.io/cluster-monitoring: "true"
+  name: openshift-special-resource-operator
```
Do you want to add manifests-k8s to the PR? If yes, please rerun `PLATFORM=k8s make manifests` as they are not fully updated.

Commit:
Added missing diffs

dagrayvid commented:
/lgtm
